misk-hibernate/src/main/kotlin/misk/hibernate/migrate/BulkShardMigrator.kt

```diff
@@ -154,6 +156,10 @@ class BulkShardMigrator<R : DbRoot<R>, C : DbChild<R, C>> private constructor(
     checkNotNull(targetRoot) { "You have to specify entity root target" }
     checkNotNull(sourceRoot) { "You have to specify entity root source" }
 
+    if (transacter.config().type == DataSourceType.TIDB) {
```

Comment:
I think we can do this on plain MYSQL as well btw? Right? Should optimize things nicely on Aurora as well! Do you want to try that in a follow up?